lib/src/install/osconfig.rs

```diff
@@ -18,7 +18,8 @@ pub(crate) fn inject_root_ssh_authorized_keys(
     // While not documented right now, this one looks like it does not newline wrap
     let b64_encoded = ostree_ext::glib::base64_encode(contents.as_bytes());
     // See the example in https://systemd.io/CREDENTIALS/
-    let tmpfiles_content = format!("f~ /root/.ssh/authorized_keys 600 root root - {b64_encoded}\n");
+    let tmpfiles_content =
+        format!("f~ /var/roothome/.ssh/authorized_keys 600 root root - {b64_encoded}\n");
```

Comment:
OK this is a complex mess.  The thing is...I would actually like to support a case where `/root` is actually immutable state i.e. it's part of the base image, not in `/var`.

(The same for `/home`; for many headless server cases, persistent home directories are actually an anti-pattern.  We do have it set up this way for OCP nodes for example, but in e.g. GKE `/home` is a tmpfs because ssh to a given node is...again more of an anti-pattern)

I guess anyone who wants to do that could of course symlink `/var/roothome -> /root` or so, but...that has a few drawbacks, among them that we want to avoid "base OS code" in `/usr` inherently trusting local mutable state in `/var` as it would need to do if it traversed that symlink.

See https://www.chromium.org/chromium-os/chromiumos-design-docs/hardening-against-malicious-stateful-data/ for some examples of this.

So...mmm...I think the best fix here is probably to query the root and generate the target pointing directly at the end state or so?

Reply:
On a distinct thread, one trap I think we have here is systemd *also* ships a default tmpfiles.d line that sources the SSH key from a credential if available...and I think we may be somewhat skating the "duplicate tmpfiles.d entries" problem there today.

Maybe in theory we should actually be injecting a "static credential" via...I guess the generic baseline would be a `systemd.set_credential` karg?

Reply:
Oh, thank you! That is important background. I will need to rethink that then, I did not consider the /root immutable case.

Reply:
More commonly people will make `/root` and `/home` a `tmpfs`; the upside of that is that it works great when writing files via tmpfiles.d as this unit does, but it also means that any content in the container image is silently hidden, which can be confusing.

Reply:
I pushed an update to do this (didn't test it beyond units) but what do you think?